### PR TITLE
tests/nested/manual: enable serial assertions on testkeys nested VM's

### PR DESCRIPTION
**Pull request by anonymouse64**
This actually works now. We didn't make any changes to fakestore to enable
this, so I think this actually worked all along and I just wasn't testing it
properly when I initially tried it out.

### CODE REVIEW

File: tests/nested/manual/core20-gadget-cloud-conf/task.yaml [hunk 1]
```diff
@@ -89,20 +89,16 @@ prepare: |
...
 
   nested_secboot_sign_gadget pc-gadget "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
   snap pack pc-gadget/ "$(nested_get_extra_snaps_path)"
 
-  # TODO: enable this when ready, currently serial assertions requests don't 
-  # work with the fakedevicesvc, needs a little bit of work somewhere
   # start fake device svc
-  # #shellcheck disable=SC2148
-  #  systemd-run --unit fakedevicesvc fakedevicesvc localhost:11029
+  #shellcheck disable=SC2148
```
Comment (bboozzoo) [inline tests/nested/manual/core20-gadget-cloud-conf/task.yaml]: do we need this shellcheck disable bit?

File: tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml [hunk 2]
```diff
@@ -82,22 +82,19 @@ prepare: |
...
   snap pack pc-gadget/ "$(nested_get_extra_snaps_path)"
   rm -rf pc-gadget/
 
   rm -f "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
 
-  # TODO: enable when ready
   # start fake device svc
-  # #shellcheck disable=SC2148
-  #  systemd-run --unit fakedevicesvc fakedevicesvc localhost:11029
+  #shellcheck disable=SC2148
```
Comment (bboozzoo) [inline tests/nested/manual/grade-signed-above-testkeys-boot/task.yaml]: same here

File: tests/nested/manual/grade-signed-cloud-init-testkeys/task.yaml [hunk 3]
```diff
@@ -78,23 +78,18 @@ prepare: |
...
   rm -rf pc-gadget/
 
   rm -f "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
 
-  # TODO: enable this when ready, currently serial assertions requests don't 
-  # work with the fakedevicesvc, needs a little bit of work somewhere
-  # start fake device svc
-  # #shellcheck disable=SC2148
-  #  systemd-run --unit fakedevicesvc fakedevicesvc localhost:11029
+  #shellcheck disable=SC2148
```
Comment (bboozzoo) [inline tests/nested/manual/grade-signed-cloud-init-testkeys/task.yaml]: this one too

File: tests/nested/manual/core20-gadget-cloud-conf/task.yaml (hunk 1 above)
Comment (bboozzoo) [reply tests/nested/manual/core20-gadget-cloud-conf/task.yaml]: I've pushed a patch fixing those

Comment (anonymouse64) [reply tests/nested/manual/core20-gadget-cloud-conf/task.yaml]: thanks for this

**Review by bboozzoo — APPROVED** [review]

**Review by mardy — APPROVED** [review]
LGTM, though I don't much about assertions so my review is mostly formal :-)

**Comment by anonymouse64** [comment]
Thanks @bboozzoo for the fixes here

File: tests/nested/manual/grade-signed-cloud-init-testkeys/task.yaml [hunk 4]
```diff
@@ -135,10 +132,6 @@ execute: |
   echo "And we can run things as the normal user"
   tests.nested exec_as normal-user ubuntu "sudo true"
 
-  # TODO: check that we got a serial assertion via the fakedevicesvc
```
Comment (mvo5) [inline tests/nested/manual/grade-signed-cloud-init-testkeys/task.yaml:138]: The other test was adding:
```
   tests.nested exec "sudo snap model --verbose --serial" | MATCH "serial:\s+7777"
```
should this be added here as well?

**Review by sergiocazzolato — APPROVED** [review]
Nice, thanks for the update

File: tests/nested/manual/grade-signed-cloud-init-testkeys/task.yaml (hunk 4 above)
Comment (anonymouse64) [reply tests/nested/manual/grade-signed-cloud-init-testkeys/task.yaml:138]: I thought it was a bit overkill to test it in multiple places actually, but it's easy enough for me to add that to this test too

Comment (anonymouse64) [reply tests/nested/manual/grade-signed-cloud-init-testkeys/task.yaml:138]: added

**Comment by anonymouse64** [comment]
The failures here are unrelated, this is ready to land

**Pull request merged**